ACMILabs/transcoder
# #19 Add restricted file handling

Resolves #19

- [x] When a video file is found by the transcoder, check to see whether the basename ends in 'RESTRICTED' and ignore if so.
- [x] Create a ticket to lint the transcoder #20

## Testing

1. Create a `dev.env` file: `cp dev.tmpl.env dev.env`
1. Add the watch folders to your filesystem... e.g.

```
SMB_WATCH=/Users/sloffler/Code/transcoder-test-folders/watch
SMB_ACCESS=/Users/sloffler/Code/transcoder-test-folders/access
SMB_WEB=/Users/sloffler/Code/transcoder-test-folders/web
SMB_MASTER=/Users/sloffler/Code/transcoder-test-folders/master
SMB_OUTPUT=/Users/sloffler/Code/transcoder-test-folders/output
```

1. Build the project: `make build`
1. Run tests: `docker exec -it transcoder_transcoder_1 make test`
1. Make a [short video file](https://acmiteam.slack.com/files/UBJ7W5H8F/F01C8RUU8BY/acmi.mp4) with the name: `B2004203_mo01_RESTRICTED_AmazingVideo.mp4`
1. Add it to your `watch` folder
1. Start the transcoder: `make up`
1. Note that it doesn't get transcoded
1. Remove `RESTRICTED` from the filename, delete the `.lock` file, and start the transcoder again: `make up`
1. Note that it does get transcoded
